cli/src/stake.rs

```diff
@@ -1419,14 +1434,17 @@ pub fn process_create_stake_account(
             return Err(CliError::BadParameter(err_msg).into());
         }
 
-        let minimum_balance =
-            rpc_client.get_minimum_balance_for_rent_exemption(StakeState::size_of())?;
+        if !force {
+            let minimum_balance = rpc_client
+                .get_minimum_balance_for_rent_exemption(StakeState::size_of())?
+                + rpc_client.get_stake_minimum_delegation()?;
```

Comment:
The way it is set up with the sanity_check_result seems like unnecessary work because it first does the check then assesses the need for the check, which results in excess RPC calls and wait time for the user, so imho we should first assess whether a check is needed, then perform it.